core/recourse_iterator.py

```diff
@@ -25,29 +24,29 @@ class RecourseIterator:
     """Class for iterating recourse."""
     def __init__(self,
                  recourse_method: RecourseMethod,
-                 preprocessor: dp.Preprocessor,
+                 adapter: recourse_adapter.RecourseAdapter,
                  certainty_cutoff: float = None,
                  check_certainty: CertaintyChecker = None):
         """Creates a new recourse iterator.
 
         Args:
             recourse_method: The recourse method to use.
-            preprocessor: The preprocessor.
+            adapter: The adapter.
```

Comment:
Nit: adapter: A RecourseAdapter object to transform the data between human-readable and embedded space.